.github/workflows/ion-java-performance-regression-detector.yml

```diff
@@ -58,15 +77,60 @@ jobs:
       - name: Build ion-java from the new commit
         run: mvn clean install
 
-      - name: Build ion-java-benchmark-cli based on the new ion-java
+      - name: Checkout ion-java-benchmark-cli
         uses: actions/checkout@v2
         with:
-          repository: amzn/ion-java-benchmark-cli
-          ref: master
-      - run: mvn clean install
+          repository: linlin-s/ion-java-benchmark-cli
+          ref: final
+
+      - name: Build ion-java-benchmark-cli based on the new ion-java
+        run: mvn clean install
 
       - name: Check the version of ion-java
         run: java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar --version
 
-      - name: Test read preformance of ion-java from the new commit
-        run: java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar read testWorkflow.ion -o readPerformanceNew.ion -r ion
\ No newline at end of file
+      - name: Create directories for test data and benchmark results
+        run: |
+          mkdir -p benchmarkResults
+          mkdir -p testData
+
+      - name: Download test Ion Data to artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: test Ion Data
+          path: testData
+
+      - name: Download benchmark results of ion-java from the existing commit to artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: Benchmark result
+          path: benchmarkResults
+
+      - name: Benchmark ion-java from the new commit and add the generated benchmark result to the existing directory
+        run: java -jar target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar organize --test-ion-data /home/runner/work/ion-java/ion-java/testData /home/runner/work/ion-java/ion-java/benchmarkResults
+
+      - name: Upload new benchmark results directory to artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: Benchmark result
+          path: benchmarkResults
+
+      - name: Detect performance regression
+        id: regression_result
+        run: |
+          cd benchmarkResults
+          result=true
+          for FILE in *; do message=$(java -jar /home/runner/work/ion-java/ion-java/target/ion-java-benchmark-cli-0.0.1-SNAPSHOT-jar-with-dependencies.jar compare --benchmark-result-previous $FILE/previous.ion --benchmark-result-new $FILE/new.ion --threshold /home/runner/work/ion-java/ion-java/tst/com/amazon/ion/benchmark/threshold.ion $FILE/report.ion | tee /dev/stderr) &&if [ "$message" != "Regression not detected" ]; then result=false;fi; done
+          echo "::set-output name=regression-result::$result"
```

Comment:
> Since we used a for loop to iterate all files, is it possible to know which file(s) fail the workflow?

Yes, the output message will contain these information, i.e. Including the test data and parameters we use which can help users to map with the benchmark results directory.